### ISSUE TITLE
Better error messages if missing Stan submodule

### ISSUE THREAD

**Issue by seantalts** [stan-dev/cmdstan#565]
#### Summary:
We'd like nice error messages if someone clones CmdStan without `--recursive`.

#### Reproducible Steps:
Clone CmdStan without the `--recursive` flag and end up missing the stan subdirectory and try to make an example.

#### Current Output:
```
Seans-MBP ~/scm/temp/cmdstan (build/c++11) $ make examples/bernoulli/bernoulli
makefile:29: stan/lib/stan_math/make/libraries: No such file or directory
makefile:34: stan/lib/stan_math/make/default_compiler_options: No such file or directory
makefile:52: stan/lib/stan_math/make/detect_cc: No such file or directory
makefile:62: stan/lib/stan_math/make/detect_os: No such file or directory
find: stan/src/stan/lang: No such file or directory
find: stan/src/stan/lang: No such file or directory
make: *** No rule to make target `stan/lib/stan_math/make/detect_os'.  Stop.
```

#### Expected Output:
Ideally something mentioning that they should make sure they update the submodules with `git submodule init; git submodule update --recursive` or to reclone with `git clone git@github.com:stan-dev/cmdstan.git --recursive`.

#### Current Version:
v2.16.0

**Comment by aornugent** [comment]
This would also be good information to include in the [wiki](https://github.com/stan-dev/cmdstan/wiki/Getting-Started-with-CmdStan)

```
git clone https://github.com/stan-dev/cmdstan.git --recursive
```

**Comment by seantalts** [comment]
@aornugent good call! I changed it. Too bad wikis don't take pull requests.

**Comment by rok-cesnovar** [comment]
This issue has since changed, as of switching to stanc3 the error when not recursively cloning is: 

```
make/stanc:11: *** Can't detect OS properly. This will impede automatically downloading the correct stanc. Please visit https://github.com/stan-dev/stanc3/releases and download a stanc binary for your OS and place it in ./bin/stanc.  Stop.
```
Which is less errors but still confusing to the user. Fix incoming in #738
